clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -156,7 +156,7 @@ public static int utf8Length(CharSequence s) {
      * Read the given byte buffer into a byte array
      */
     public static byte[] toArray(ByteBuffer buffer) {
-        return toArray(buffer, 0, buffer.limit());
```

Comment:
The `toArray` we're delegating to begins from `buffer.position()`, so `remaining()` seems correct. I'll add a test case to figure it out.